# Calculating sales and Cost of Goods Sold (COGS)

_Financial Forecasting in Python_
**Lesson:** 1-2. Calculating sales and Cost of Goods Sold(COGS)

📄 Notes: [02 - Calculating sales and Cost of Goods Sold (COGS)](docs/02-Calculating%20sales%20and%20Cost%20of%20Goods%20Sold(COGS).md)

## Overview

Compute revenue (sales) and COGS, then derive gross profit.

## Learning objectives

- Calculate forecast sales from unit price and volume, including a weighted average price for a sales mix.
- Adjust a sales forecast for discounts across different product ratios.
- Calculate Cost of Goods Sold (COGS) from fixed and variable costs, and derive the cost per unit.
- Compute the break-even point in units from fixed costs, price per unit, and variable cost per unit.

## Setup

In [1]:
import sys; sys.path.insert(0, '../../src')

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

from ffp import sample_prices, error_report  # shared helpers

## Walkthrough

In [2]:
# TODO: lesson code here
# prices = sample_prices()
# prices.plot(figsize=(10, 4), title='Sample price series')
# plt.show()

## Exercise

**Excersise 1 - Calculating sales**

To calculate a sales forecast we need to know both the price we wish to sell our product and the number of units we expect to sell. In this example we will have a look at a company T-Z producing and selling T-Shirts.

The company offers its customers two options - a basic shirt or a customized shirt. A T-shirt costs 15 USD for a basic shirt, and 25 USD for a customized one.

T-Z have forecast orders for T-Shirts for delivery next month, and want to use the same sales mix of 60% basic and 40% customized. As this mix is relatively stable, we can calculate an average sales price per unit for T-Z.

In [3]:
# Set variables units sold and sales price of the T-shirts (basic and custom)
salesprice_basic = 15
salesprice_custom = 25

# Calculate the combined sales price taking into account the sales mix
average_sales_price = (salesprice_basic * 0.6) + (salesprice_custom * 0.4)

forecast_units = 500

# Calculate the total sales for next month
sales_USD = forecast_units * average_sales_price

# Print the total sales
print("Next month's forecast sales figure is {:.2f} USD.".format(sales_USD))

Next month's forecast sales figure is 9500.00 USD.


**Exercise 2 - Forecasting sales with a discount**

Sales are often made with special mixes and discounts. Discounts are a great marketing tool and a way to move stock. Let's look again at company T-Z.

T-Z has launched a new range of T-Shirts linked to a celebrity meme for 40 USD per T-Shirt. They have excess stock of Celebshirt1 on their shelves, and in expectation of the release of Celebshirt2 in February, they have announced a 40% discount on Celebshirt1 in February.

For this exercise, prices are in USD, and unit amounts are total units sold in the respective month. The following variables have been defined for you:

sales_price = 40

units_january = 500

units_february = 700

The January sales only include sales of Celebshirt1.

The February sales include sales of Celebshirt1 and Celebshirt2, at a ratio of 45:55.

In [4]:
sales_price = 40
units_january = 500
units_february = 700

# Forecast the sales of January
sales_january = sales_price * units_january

# Forecast the discounted price
dsales_price = sales_price * 0.6

# Forecast the sales of February
sales_february = (sales_price * units_february * 0.55) + (dsales_price * units_february * 0.45)

# Print the forecast sales for January and February
print("The forecast sales for January and February are {} and {} USD respectively.".format(sales_january, sales_february))

The forecast sales for January and February are 20000 and 22960.0 USD respectively.


**Exercise 3 - Calculating COGS**

Cost of Goods Sold (COGS) is calculated as follows:

$$\text{COGS} = \text{Opening value of stock} - \text{Closing value of stock} + \text{Fixed costs current period} + \text{Variable cost current period}$$

We assume no closing or opening balances, stock is sold in the same month that it is produced.

Material costs to produce one T-shirt is 8 USD. Labor costs are 2 USD per shirt. As you only use materials and labor to make the shirt, this is known as variable costs. The costs to rent a machine that produces these shirts is 1300 USD per month, regardless of the amount of shirts produced. This is therefore an example of a fixed cost, as fixed costs in COGS are costs directly related to the product but are incurred whether you produce the products or not.

In [5]:
machine_rental = 1300
material_costs_per_unit = 8
labor_costs_per_unit = 2
units_jan = 200
units_feb = 250

# Set the variables for fixed costs and variable costs
fixed_costs = machine_rental
variable_costs_per_unit = material_costs_per_unit + labor_costs_per_unit

# Calculate the cogs for January and February
cogs_jan = (units_jan * variable_costs_per_unit) + fixed_costs
cogs_feb = (units_feb * variable_costs_per_unit) + fixed_costs

fixed_costs = machine_rental
variable_costs_per_unit = material_costs_per_unit + labor_costs_per_unit
cogs_jan = (units_jan * variable_costs_per_unit) + fixed_costs
cogs_feb = (units_feb * variable_costs_per_unit) + fixed_costs

# Calculate the unit cost for January and February
unit_cost_jan = cogs_jan / units_jan
unit_cost_feb = cogs_feb / units_feb

# Print the January and February cost per unit
print("The cost per unit for January and February are {} and {} USD respectively.".format(unit_cost_jan, unit_cost_feb))

The cost per unit for January and February are 16.5 and 15.2 USD respectively.


**Exercise 4 - Calculating the break-even point**

Tracking and understanding how cost per unit varies by month is critical, as the COGS influence the gross profit. At a certain point, if we do not make enough sales in the month to cover our costs, we will enter a loss situation.

How do we know we have produced enough units to sell to cover our costs? We do this by using a concept called the break-even point. The formula is as follows:

$$\text{Break-even Point (Units)} = \frac{\text{Fixed costs}}{\text{Price per unit} - \text{Variable cost per unit}}$$

In the last exercise, we calculated a unit cost of 16.5 USD for January and 15.2 USD for February. Keep this in mind when calculating the break-even point.


In [6]:
fixed_costs = 1300
variable_costs_per_unit = 10
units_jan = 200
units_feb = 250
sales_price = 16
cogs_jan = (units_jan * variable_costs_per_unit) + fixed_costs
cogs_feb = (units_feb * variable_costs_per_unit) + fixed_costs

# Calculate the break-even point (in units) for Wizit
break_even = fixed_costs/(sales_price - variable_costs_per_unit)

# Print the break even point in units
print("The break even point is {} units.".format(break_even))

# Forecast the gross profit for January and February
gross_profit_jan = (sales_price*units_jan) - cogs_jan
gross_profit_feb = (sales_price*units_feb) - cogs_feb

# Print the gross profit for January and February
print("The gross profit for January and February are {} and {} USD respectively.".format(gross_profit_jan, gross_profit_feb))

The break even point is 216.66666666666666 units.
The gross profit for January and February are -100 and 200 USD respectively.


## Summary

Recap and  key points see in lessons/01-income-statements/docs/02-Calculating sales and Cost of Goods Sold(COGS).md